src/Processors/Formats/Impl/CHColumnToArrowColumn.h

```diff
@@ -34,6 +34,23 @@ class CHColumnToArrowColumn
     };
 
     CHColumnToArrowColumn(const Block & header, const std::string & format_name_, const Settings & settings_);
+    CHColumnToArrowColumn(const ColumnsWithTypeAndName & header_columns_, const std::string & format_name_, const Settings & settings_);
+
+    /// Makes a copy of this converter.
+    /// This can be useful to prepare for conversion in multiple threads.
+    std::unique_ptr<CHColumnToArrowColumn> clone(bool copy_arrow_schema = false) const;
+
+    /// Initializes the arrow schema.
+    /// It's not done in the constructor because LowCardinality column from header always has indexes type UInt8,
+    /// so we should get proper indexes type from the first chunk of data.
```

Comment:
Cannot the type of indexes change between different chunks of data? I am really not sure and didn't dig into the code, but based on the description of `low_cardinality_use_single_dictionary_for_part` we can have either multiple small dictionaries, or one big dictionary. I guess we use small index type first, then increase the size of the index on overflow.

Reply:
> I guess we use small index type first, then increase the size of the index on overflow.

Yes, that's true. However for the Arrow format we have to use the same arrow schema for all chunks. So we initialize the arrow schema from the first chunk and hope it will be OK for other chunks too. It's not a big problem actually because (see [getArrowTypeForLowCardinalityIndexes](https://github.com/ClickHouse/ClickHouse/blob/af36708a8cfd3155e2ca4e51f132b699285a9438/src/Processors/Formats/Impl/CHColumnToArrowColumn.cpp#L896-L922)) we always use at least 32 bits for arrow indices which should enough for most cases.